### DISCUSSION TITLE
Passing state that lives after a redirect happened

### DISCUSSION THREAD

**Discussion by dhardtke** [oakserver/oak#292] [Q&A]
Hi,

I would like to use the `state` object to pass information to a route after a redirect happened about a previous action, e.g.:

```ts
import { Application, Router } from "https://deno.land/x/oak/mod.ts";

const router = new Router();
router
  .get("/", (context) => {
    if (context.state.flash === "postSuccessful") {
        context.response.body = "Post was sucessful!";
    } else {
        context.response.body = "Hello world!";
    }
  })
  .post("/create", (context) => {
    context.state.flash = "postSuccessful";
    ctx.response.redirect("/");
  });

const app = new Application({state: {flash: ""}});
app.use(router.routes());
app.use(router.allowedMethods());

await app.listen({ port: 8000 });
```

but since the redirect leads Oak to creating a new Context this does not work.

Does anyone have an idea on how to temporarily pass some data between routes? I thought of the ugly solution of creating some sort of Proxy object that acts as a "Box<T>" which clears its value after retrieving it which could be passed to the App as initial state but this is quite ugly...

Thanks.

**Comment by kitsonk** [comment]
`context.app.state` will persist between requests/contexts.  You just need to realise it persists for the whole of life of the application.

**Reply by dhardtke** [reply]
Thanks, but unfortunately this is not really what I am looking for or it's not working as expected:

I'd like to pass some information to the next route (e.g. "Item has been saved successfully") that gets cleared by the route after it is read.

Once I do a redirect the state change is no longer visible. Using `context.app.state` or `context.state` does not make any difference whatsoever.
